Peoplesoft/main.tf

```diff
@@ -0,0 +1,538 @@
+            destination_port_ranges = "9200"
+            destination_address_prefix = "*"             
+        },
+        {
+            source_port_ranges =  "*" 
+            source_address_prefix = "${local.subnetPrefixes["bastion"]}"                
+            destination_port_ranges = "22" 
+            destination_address_prefix = "*"             
+        }
+    ]
+
+    elasticsearch_sr_outbound = [
+                {
+            source_port_ranges =  "*" 
+            source_address_prefix = "${local.subnetPrefixes["application"]}"  # ob to Application Servers               
+            destination_port_ranges = "9033-9039" 
+            destination_address_prefix = "*"             # Need to support ASG
+        },
+        {
+            source_port_ranges =  "*" 
+            source_address_prefix = "${local.subnetPrefixes["webserver"]}"  # ob to WebServers            
+            destination_port_ranges = "8000" 
+            destination_address_prefix = "*"            
+        }
+    ]
+
+        toolsclient_sr_inbound = [
+        {
+            source_port_ranges =  "*" 
+            source_address_prefix = "${local.subnetPrefixes["application"]}"             
+            destination_port_ranges = "5985-5986"
+            destination_address_prefix = "*"             
+        },
+        {
+            source_port_ranges =  "*" 
+            source_address_prefix = "${local.subnetPrefixes["bastion"]}"                
+            destination_port_ranges = "22" 
+            destination_address_prefix = "*"             
+        }
+    ]
+
+    toolsclient_sr_outbound = [
+
+    ]
+    
+    database_sr_inbound = [
+        {
+            source_port_ranges =  "*" 
+            source_address_prefix = "${local.subnetPrefixes["application"]}"                 
+            destination_port_ranges =  "1521" 
+            destination_address_prefix = "*"             
+        },
+        {
+            source_port_ranges =  "*" 
+            source_address_prefix = "${local.subnetPrefixes["bastion"]}"  # input from the Load Balancer only.            
+            destination_port_ranges =  "22" 
+            destination_address_prefix = "*"                
+        }
+    ]
+    database_sr_outbound = [
+    ]
+
+    ##########################################
+    ## VMs in these subnets will need Availability sets.
+    ##########################################
+  #  needAVSets = [ "presentation", "application" ]
+}
+
+############################################################################################
+# Create a resource group
+resource "azurerm_resource_group" "rg" {
+  name     = "${var.resource_group_name}"
+  location = "${var.location}"
+  tags     = "${var.tags}"     
+}
+
+############################################################################################
+# Create the virtual network 
+
+resource "azurerm_virtual_network" "vnet" {
```

Comment:
Check if the VNET exists using data source so we don't recreate it:

data "azurerm_virtual_network" "primary_vnet" {
    name = "${var.vnet_name}"
    resource_group_name = "${var.vnet_resource_group_name}"
    count = "${var.vnet_cidr == "0" ? 1 : 0}"
}

Setup local variables to determine whether to pickup the VNET name and CIDR from data source or resource:

 vnet_name = "${var.vnet_cidr == "0" ? 
    element(concat(data.azurerm_virtual_network.primary_vnet.*.name, list("")), 0) :
    element(concat(azurerm_virtual_network.primary_vnet.*.name, list("")), 0)}"

  vnet_cidr = "${var.vnet_cidr == "0" ? element(data.azurerm_virtual_network.primary_vnet.address_space, 0) : var.vnet_cidr}"

This is because in our 'InterConnect' TF, we're already creating the VNET.

Reply:
and then any time you need to call vnet_name (for example), you'd call it as local.vnet_name rather than the var value.

Reply:
Done